commit 81984a3637e19582762af618532c5de39a68f0a0 (refs/changes/84/584/1)
Author: Trent Mick <trentm@gmail.com>
Date:   2016-10-03T16:36:26-07:00 (3 years ago)
    
    TOOLS-1558 TOOLS-1555 broke sdcnode builds: 'npm install' isn't run, buildEnv changes break on spaces

diff --git a/Makefile b/Makefile
index e5185e6..3f03844 100644
--- a/Makefile
+++ b/Makefile
@@ -42,8 +42,11 @@ nodesrc: | build/src
 	cd build/src && git checkout master \
 		&& git fetch origin && git pull --rebase origin master
 
+node_modules/.bin/semver:
+	npm install
+
 .PHONY: nodes
-nodes: nodesrc
+nodes: nodesrc node_modules/.bin/semver
 	./tools/build-all-nodes $(TOP)/build/nodes $(STAMP) "this.image=='$(HOST_IMAGE)'"
 
 .PHONY: bits
diff --git a/package.json b/package.json
index e5440b3..f639040 100644
--- a/package.json
+++ b/package.json
@@ -1,14 +1,13 @@
 {
   "name": "sdcnode",
   "description": "builds of node for SDC",
-  "version": "1.0.0",
+  "version": "1.0.1",
   "private": true,
   "author": "Joyent",
   "engines": [
-    "node >=0.6.0"
+    "node >=0.10.0"
   ],
   "devDependencies": {
-    "async": "0.1",
     "semver": "5.3.0"
   },
   "license": "MPL-2.0"
diff --git a/tools/build-a-node b/tools/build-a-node
index af9788d..66b68a4 100755
--- a/tools/build-a-node
+++ b/tools/build-a-node
@@ -122,8 +122,11 @@ fi
 ./configure $configureFlags --prefix=$buildDir/node
 
 buildEnv=$(echo "$buildOpts" | json build_env)
-make V=1 CC=$GCC CXX=$GXX $buildEnv
-make V=1 CC=$GCC CXX=$GXX $buildEnv install
+if [[ -n "$buildEnv" ]]; then
+    eval "export $buildEnv"
+fi
+make V=1 CC=$GCC CXX=$GXX
+make V=1 CC=$GCC CXX=$GXX install
 
 # Optionally include GCC runtime libs. If a version of GCC was used to compile
 # node is greater than the version of the GCC runtime libs included in the
